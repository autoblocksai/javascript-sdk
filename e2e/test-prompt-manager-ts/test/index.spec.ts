import { promptManager, main, PromptTrackingId } from '../index';

jest.mock('@autoblocks/client');
jest.setTimeout(30000);

describe('main', () => {
  it('builds prompts for feature A', async () => {
    await main();

    promptManager.snapshots(PromptTrackingId.A).forEach((snapshot) => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  it('builds prompts for feature B', async () => {
    await main();

    promptManager.snapshots(PromptTrackingId.B).forEach((snapshot) => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
