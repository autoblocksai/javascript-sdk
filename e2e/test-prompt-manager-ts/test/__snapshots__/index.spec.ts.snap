// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoblocksPromptBuilder handles nested templates 1`] = `"I am I am the value!"`;

exports[`AutoblocksPromptBuilder handles placeholders 1`] = `
"It should work with placeholders that have no spaces:
I am the placeholder value

With spaces:
I am the placeholder value

Camel case:
I am the camel case value

Snake case:
I am the snake case value

Invalid JS names:
I am not a valid JS variable name but that is ok"
`;
